cmake/defaults/Packages.cmake

```diff
@@ -268,19 +273,24 @@ endif()
 
 if (PXR_ENABLE_MATERIALX_SUPPORT)
     find_package(MaterialX REQUIRED)
-    add_definitions(-DPXR_MATERIALX_SUPPORT_ENABLED)
+    list(APPEND PXR_FEATURE_DEFINITIONS PXR_MATERIALX_SUPPORT_ENABLED)
 endif()
 
 if(PXR_ENABLE_OSL_SUPPORT)
     find_package(OSL REQUIRED)
     set(REQUIRES_Imath TRUE)
-    add_definitions(-DPXR_OSL_SUPPORT_ENABLED)
+    list(APPEND PXR_FEATURE_DEFINITIONS PXR_OSL_SUPPORT_ENABLED)
 endif()
 
 if (PXR_BUILD_ANIMX_TESTS)
     find_package(AnimX REQUIRED)
 endif()
 
+# Add the definitions globally. These end up in the target
+# COMPILE_DEFINITIONS property, but not in the INTERFACE_COMPILE_DEFINITIONS,
+# which means that they are not exported targets. We have to add them manually.
+add_compile_definitions(${PXR_FEATURE_DEFINITIONS})
```

Comment:
How hard would it be to support `INTERFACE_COMPILE_DEFINITIONS` on OpenUSD targets ?
This seems like a more robust solution as we would get those transitively, and we can introspect those by querying CMake target properties.

Reply:
I agree that this would be a nicer solution, and I tried at first to add a single top-level `INTERFACE` library to contain the compile definitions. And while I could get it to build, I couldn't figure out the export. It never showed up in the targets file. Pixar does a lot of wrapping around the `add_library` code, the logic is complex, and I had a hard time debugging the issue, so I abandoned that idea for something much more simple...